contracts/pm/TicketBroker.sol

```diff
@@ -57,35 +57,31 @@ contract TicketBroker {
         _;
     }
 
-    modifier processDeposit(address _sender, uint256 _amount) {
-        Sender storage sender = senders[_sender];
-        sender.deposit += _amount;
-        if (_isUnlockInProgress(sender)) {
-            _cancelUnlock(sender, _sender);
-        }
-
-        _;
-
-        emit DepositFunded(_sender, _amount);
+    constructor(uint256 _minPenaltyEscrow, uint256 _unlockPeriod) internal {
+        minPenaltyEscrow = _minPenaltyEscrow;
+        unlockPeriod = _unlockPeriod;
     }
 
-    modifier processPenaltyEscrow(address _sender, uint256 _amount) {
-        require(_amount >= minPenaltyEscrow, "penalty escrow amount must be >= minPenaltyEscrow");
-
-        Sender storage sender = senders[_sender];
-        sender.penaltyEscrow += _amount;
-        if (_isUnlockInProgress(sender)) {
-            _cancelUnlock(sender, _sender);
-        }
-
-        _;
+    function fundDeposit() external payable {
+        _fundDeposit(msg.value);
+    }
 
-        emit PenaltyEscrowFunded(_sender, _amount);
+    function fundPenaltyEscrow() external payable {
+        _fundPenaltyEscrow(msg.value);
     }
 
-    constructor(uint256 _minPenaltyEscrow, uint256 _unlockPeriod) internal {
-        minPenaltyEscrow = _minPenaltyEscrow;
-        unlockPeriod = _unlockPeriod;
+    function fundAndApproveSigners(
+        uint256 _depositAmount,
+        uint256 _penaltyEscrowAmount,
+        address[] _signers
+    )
+        external
+        payable
+        checkDepositPenaltyEscrowETHValueSplit(_depositAmount, _penaltyEscrowAmount)
+    {
+        _fundDeposit(_depositAmount);
```

Comment:
Noting that now that we are not using modifiers, we no longer adhere to the checks -> state updates -> external calls pattern here since `_fundDeposit()` and `_fundPenaltyEscrow()` potentially include external calls depending on the override behavior of `processDeposit()` and `processPenaltyEscrow()`.

Not following the pattern for this wrapper function might be ok as long as we remain cautious of reentrancy attacks. This is not a problem for ETHTicketBroker (which does not make any external calls) and LivepeerETHTicketBroker (since the external call is to a trusted contract - namely the Minter contract that is part of the Livepeer contract system). This might be a problem for ERC20TicketBroker if a developer deployed the contract and pointed it to an untrusted token contract.

In addition to no longer using the checks -> state updates -> external calls pattern, this wrapper function can end up executing 2 external calls when it only needs to execute 1. For example, in LivepeerETHTicketBroker, this function will execute `minter().trustedDeposit.value()` twice, once for `_depositAmount` and once for `_penaltyEscrowAmount`. Instead, this function could execute `minter().trustedDeposit.value()` once with `_depositAmount + _penaltyEscrowAmount` which was the original behavior when we were using modifiers. This is also the case for ERC20TicketBroker which will execute 2 ERC20 `transferFrom()` when it only needs to execute 1.

Reply:
Agreed and tried to resolve it all in the latest commit https://github.com/livepeer/protocol/pull/264/commits/4c06692eb7bdb38407f80a4b64a3963bf9293e84

Want to check it out?